File: 2779.maximum-beauty-of-an-array-after-applying-operation.cpp
/*
 * @lc app=leetcode id=2779 lang=cpp
 *
 * [2779] Maximum Beauty of an Array After Applying Operation
 */

// @lc code=start
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
class Solution {
public:
    int maximumBeauty(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int maxBeauty = 0;
        int l = 0, r = 0;

        for (;r < nums.size(); r++) {
            while (nums[r] - nums[l] > 2 * k) {
                l++;
            }
            maxBeauty = max(maxBeauty, r - l + 2);
        }
        return maxBeauty;
        
    }
};
// @lc code=end

File: 2779.maximum-beauty-of-an-array-after-applying-operation.py
#
# @lc app=leetcode id=2779 lang=python3
#
# [2779] Maximum Beauty of an Array After Applying Operation
#

# @lc code=start
from typing import List
class Solution:
    def maximumBeauty(self, nums: List[int], k: int) -> int:
        nums.sort()
        left, right = 0, 0
        max_len = 0

        while right < len(nums):
            
            while nums[right] - nums[left] > 2 * k:
                left += 1
            max_len = max(max_len, right - left + 1)
            right += 1
        return max_len
            

# @lc code=end

